lib/src/transformers.dart

```diff
@@ -284,3 +284,50 @@ String? toTimestampString(String? value) {
   }
   return null;
 }
+
+Map<String, dynamic> getEnrichedPayload(dynamic payload) {
+  final postgresChanges = payload['data'] ?? payload;
+  final schema = postgresChanges['schema'];
+  final table = postgresChanges['table'];
+  final commitTimestamp = postgresChanges['commit_timestamp'];
+  final type = postgresChanges['type'];
+  final errors = postgresChanges['errors'];
+
+  final enrichedPayload = {
+    'schema': schema,
+    'table': table,
+    'commit_timestamp': commitTimestamp,
+    'eventType': type,
+    'new': {},
+    'old': {},
+    'errors': errors,
+  };
+
+  return {
+    ...enrichedPayload,
+    ...getPayloadRecords(postgresChanges),
+  };
+}
+
+Map<String, Map<String, dynamic>> getPayloadRecords(dynamic payload) {
+  final records = <String, Map<String, dynamic>>{
+    'new': {},
+    'old': {},
+  };
+
+  if (payload['type'] == 'INSERT' || payload['type'] == 'UPDATE') {
+    records['new'] = convertChangeData(
+      List<Map<String, dynamic>>.from(payload['columns']),
+      Map<String, dynamic>.from(payload['record']),
+    );
+  }
+
+  if (payload?['type'] == 'UPDATE' || payload?['type'] == 'DELETE') {
```

Comment:
Is there a reason adding `?` here, but not above?

Reply:
Thanks for catching this. I have added to the ones above as well.

Reply:
But payload is not null in `getEnrichedPayload`, so I would change the type of payload to `Map<String, dynamic>` and remove both `?` instead.

Reply:
Yeah, true. There are a lot of loose ends when it comes to typing in this library. It might be worth tightening it up now that we have some mock tests!